Refactor refactored interaction handling

Anclagen opened the pull request:
I built this ontop of #75 I've take the general gist from it, streamlined it a little in places and added support for message triggers too.

```bash
events/
├── interaction/
│   ├── index.js                     # Main InteractionCreate dispatcher
│   │
│   ├── handlers/                    # Routing only (controllers)
│   │   ├── command.js               # Routes slash commands
│   │   ├── button.js                # Routes button interactions
│   │   ├── modal.js                 # Routes modal submissions
│   │   ├── select.js                # Routes select menus
│   │   └── handleEvent.js           # Shared routing utility
│   │
│   └── features/                    # business logic mix with discord reply & UI handling, it ain't pure!
│       ├── users/
│       │   ├── create-user-modal.js
│       │   └── handle-user-modal.js
│       │
│       ├── projects/
│       │   ├── create-project-modal.js
│       │   ├── delete-project-button.js
│       │   └── list-projects.js
│       │
│       └── session/
│           ├── handle-clock-in-button.js
│           ├── handle-clock-in-modal.js
│           ├── handle-clock-out-button.js
│           └── handle-timer-buttons.js
│
└── message/
    ├── index.js                     # Main MessageCreate dispatcher
    ├── triggers/
    │   ├── imBot.js
    │   ├── imDad.js
    │   ├── ollama.js
    │   └── ...
    └── triggers.js                  # Aggregate trigger array
 ```

fmSand committed:
refactor(interactions): refactor interaction handling with modular structure

Anclagen committed:
refactor(events): Change the handlers and added message trigger hander

Anclagen committed:
refactor: re-arranging files for better structure

Anclagen committed:
fix: triggers in the interaction event handler

fmSand commented:
Very nice improvements sir. I might have to make some changes/addons to `handleEvent` for my timer buttons as they are dynamic, but I'll do some proper testing with that branch later.  I'm good with merging this now, if you are👍  

(We can focus on the context later as its own PR)

Anclagen commented:
I was thinking for the handleEvent, I will add like I said with the context adding, perhaps make another utility that takes the context config and interaction and appends so the madness of that happens else where to avoid bloat. Also I think I need to split the event id strings, as some of my buttons on the project management are like `delete_project:<uuid>`. So if we agree to use `:` as the divide for the id name and id of the project/user/session etc.. then that should solve those problems.

Anclagen committed:
feat: Context adding and id parsing

Anclagen commented:
I've added basic context handling, the appendContext function can be the messy place where we expand what we want to add, perhaps that will be broken down into more specific functions if we need too much context later. For the moment the imports for the handlers should look something like so:
```js
import { handleCreateUserButton } from "../features/users/create-user-modal.js";

export const buttonHandlers = {
  create_user_modal_button: {
    run: handleCreateUserButton,
    context: { needsUser: true, needsSession: false },
  },
};
```

The handleEvent function will handle event ids with other ids attached relating to the project, user or session you updating, by splitting them based on `:` and taking the first part as the key.
```js
import { appendContext } from "../../../utils/appendContext.js";

export async function handleEvent(interaction, handlers) {
  const key = interaction.customId.split(":")[0];

  const handler = handlers[key];
  if (!handler) {
    console.warn(`Unhandled interaction: ${key}`);
    return;
  }

  await appendContext(interaction, handler.context);

  try {
    return await handler.run(interaction);
  } catch (err) {
    console.error(`Interaction error (${key}):`, err);
  }
}
```

If your happy with approach, approve away. I will convert the other PRs to drafts while we update them to use this.

fmSand reviewed (approved):
I approve this code 👍
